crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -212,7 +213,9 @@ impl<'db> SemanticIndexBuilder<'db> {
         let existing_definition = self
             .definitions_by_node
             .insert(definition_node.key(), definition);
-        debug_assert_eq!(existing_definition, None);
+        if existing_definition.is_some() {
+            tracing::warn!("Existing definition was unexpectedly evicted");
```

Comment:
There are a bunch of warnings I added just to try and get things working.

In total there are 228 warnings (of which this one is only hit 5 times), so it's pretty surmountable all things considered